akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcProjectionImpl.scala

```diff
@@ -443,22 +449,27 @@ private[projection] object R2dbcProjectionImpl {
       .via(handler)
   }
 
+  /**
+   * This replay mechanism is used by GrpcReadJournal
+   */
   private def triggerReplayIfPossible[Offset, Envelope](
       sourceProvider: SourceProvider[Offset, Envelope],
       offsetStore: R2dbcOffsetStore,
-      envelope: Envelope): Boolean = {
+      envelope: Envelope)(implicit ec: ExecutionContext): Future[Boolean] = {
```

Comment:
additional change in https://github.com/akka/akka-projection/pull/1255/commits/ebfec44005e4272c37bd816403c3c448bae09f04
`offsetStore.storedSeqNr` returns Future to be able to load from db